server.py

```diff
@@ -1,92 +1,108 @@
 #!/usr/bin/env python3
 
-from bottle import (
-    get, post, run, static_file, template, request, response, redirect
-)
-
-from base64 import urlsafe_b64decode as b64dec
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography.hazmat.primitives.asymmetric import rsa, padding
+from base64 import urlsafe_b64decode
 import json
 from os import getenv
-from random import getrandbits
 import re
+from urllib import parse, request
 from time import time
+from wsgiref import simple_server
+import os, mimetypes
 
-SECRET = getenv('COOKIE_SECRET', '%x' % getrandbits(128))
+DIR = os.path.dirname(os.path.abspath(__file__))
 
 META = {
-    'LA_HOSTNAME': 'laoidc.herokuapp.com',
+    'LA_ORIGIN': 'https://laoidc.herokuapp.com',
     'RP_ORIGIN': 'http://localhost:%s' % getenv('PORT', '8080'),
 }
 
 if getenv('HEROKU_APP_NAME'):
     META['RP_ORIGIN'] = 'https://%s.herokuapp.com' % getenv('HEROKU_APP_NAME')
 
 
-@get('/')
-def index():
-    # TODO: Move session handling to middleware
-    # TODO: Check cookie expiration, once added
-    session = request.get_cookie('session', secret=SECRET)
-    if session is None:
-        session = {'email': None}
+def template(tpl, status=200, **vars):
+    with open(os.path.join(DIR, tpl) + '.tpl') as f:
+        src = f.read()
+    for k, v in vars.items():
+        src = src.replace('{{ %s }}' % k, v)
+    headers = {'Content-Type': 'text/html; charset=utf-8'}
+    return 200, headers, src.encode('utf-8')
 
-    return template('template/index',
-                    email=session['email'],
-                    META=META)
 
+def index(env):
+    return template('template/index', **META)
 
-@get('/login')
-def login_get():
-    redirect('/')
 
+def login(env):
 
-@post('/login')
-def login_post():
-    jwt = request.forms.get('id_token')
+    if env['REQUEST_METHOD'] == 'GET':
+        return index(env)
```

Comment:
Clarity of purpose? If you want to see the index content, that lives on `GET /`. If you want to log in, that lives on `POST /login`.